client/components/marketing-survey/cancel-purchase-form/index.jsx

```diff
@@ -1141,11 +1143,18 @@ class CancelPurchaseForm extends React.Component {
 	}
 
 	componentDidMount() {
-		this.initSurveyState();
+		loadExperimentAssignment( 'fullscreen_precancellation_survey' ).then(
```

Comment:
It looks like you don't account for loading state, this means while it is loading a user will experience the control experience causing crossover of experience which can mess with the experiment.

Reply:
It also looks like this experiment is only run for users with `isPlan( purchase )`, if so you should fetch the assignment conditionally on that, increasing the experiment sensitivity:

```suggestion
		isPlan( purchase ) && loadExperimentAssignment( 'fullscreen_precancellation_survey' ).then(
```

Reply:
You should also run `loadExperimentAssignment` before this component mounts - as soon as the button is pressed - to prevent the loading state from occurring.

Reply:
The `CancelPurchaseForm` component is actually rendered before it gets visible to users: 

https://github.com/Automattic/wp-calypso/blob/8fbac3663b8229fa40ecd0bcfd74fa98fe6d6306/client/me/purchases/cancel-purchase/button.jsx#L290-L300

Experiment data should be already loaded once users clicks on the button, since that only switches the value of the `isVisible` prop.

Reply:
> you should fetch the assignment conditionally on that

Good idea, done in 7b50cb4.

Reply:
> Experiment data should be already loaded once users click on the button, since that only switches the value of the isVisible prop.

Yep that works :)